cli_tools/common/utils/logging/firelog/logger.go

```diff
@@ -0,0 +1,358 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package firelog
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	daisyutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"github.com/google/uuid"
+	"github.com/minio/highwayhash"
+)
+
+var (
+	httpClient            httpClientInterface = &http.Client{Timeout: 5 * time.Second}
+	serverURL                                 = decryptString(serverURLProdP1, serverURLProdP2)
+	key                                       = decryptString(keyP1, keyP2)
+	serverLogEnabled                          = false
+	logMutex                                  = sync.Mutex{}
+	nextRequestWaitMillis int64
+)
+
+// constants used by logging
+const (
+	ImageImportAction    = "ImageImport"
+	ImageExportAction    = "ImageExport"
+	InstanceImportAction = "InstanceImport"
+
+	// These strings should be interleaved to construct the real URL. This is just to (hopefully)
+	// fool github URL scanning bots.
+	serverURLProdP1 = "hts/frbslgigp.ogepscmv/ieo/eaylg"
+	serverURLProdP2 = "tp:/ieaeogn-agolai.o/1frlglgc/o"
+	keyP1           = "AzSCO1066k_gFH2sJg3I"
+	keyP2           = "IaymztUIWu9U8THBeTx"
+
+	targetSizeGb  = "target-size-gb"
+	sourceSizeGb  = "source-size-gb"
+	statusStart   = "Start"
+	statusSuccess = "Success"
+	statusFailure = "Failure"
+)
+
+type logResult string
+
+const (
+	failedOnCreateRequest          logResult = "FailedOnCreateRequest"
+	failedOnCreateRequestJSON      logResult = "FailedOnCreateRequestJSON"
+	failedToParseResponse          logResult = "FailedToParseResponse"
+	failedAfterRetry               logResult = "FailedAfterRetry"
+	failedOnUndefinedResponse      logResult = "FailedOnUndefinedResponse"
+	failedOnMissingResponseDetails logResult = "FailedOnMissingResponseDetails"
+	serverLogDisabled              logResult = "ServerLogDisabled"
+)
+
+type httpClientInterface interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func decryptString(p1, p2 string) string {
```

Comment:
I don't know about go naming conventions, but this is more like deinterleaving rather than decrypting.